### PR TITLE
Feature/designer api

### PR DESCRIPTION
**Pull request by craxkumar**
added designer api
integrated keycloak/db(mongodb)
added docker files for keycloak/mongodb

### CODE REVIEW

File: designer/controller.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
+const { nanoid } = require("nanoid");
+const utility = require('../utility/core');
+const jdlConverter = require('../utility/jsonToJdl');
+const exec = require("child_process").exec;
+const blueprintDao = require('../dao/blueprintDao');
+
+exports.test = function (req, res) {
+    return res.status(200).send({ message: "Test API for wda server" });
+};
```
Comment (rohithteja-cmi) [inline designer/controller.js:9]: test code must be cleaned up

File: designer/controller.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+/**
+ * Get specific blueprint with given project Id
+ * @param {*} req 
+ * @param {*} res 
+ */
+exports.getBlueprint = function (req, res) {
+  blueprintDao.getByProjectId({project_id: req.params.project_id})
+  .then(result => {
+    console.log("Retrieved blueprint:", result);
```
Comment (rohithteja-cmi) [inline designer/controller.js:20]: lets log only userId, objectId & projectId only. Complete result might bloat logs.

File: designer/controller.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+ */
+exports.generate = function (req, res) {
+    const body = req.body;
+    // console.log(body)
+
+    console.log(
+        "Hi",
+        "Your project Name is",
+        body.projectName
+    );
```
Comment (rohithteja-cmi) [inline designer/controller.js:64]: logs can be standardised, `console.log("Generating project: " + body.projectName + ", for user: " + userId)`

File: utility/jsonToJdl.js [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        }
+        if (applications[i].serverPort === undefined || applications[i].serverPort === "") {
+            applicationErrorList.push("Server Port cannot be empty");
+        }
+
+        // return error response
+        if (applicationErrorList.length > 0) {
+            applicationError[i] = applicationErrorList;
+            console.log(applicationError);
+            return res.status(406).send(applicationError);
```
Comment (rohithteja-cmi) [inline utility/jsonToJdl.js:57]: can be 400 Bad Request

File: utility/jsonToJdl.js [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+        if (deployment.kubernetesUseDynamicStorage !== undefined && deployment.kubernetesUseDynamicStorage === true
+            && (deployment.kubernetesStorageClassName === undefined || deployment.kubernetesStorageClassName === ""
+                || deployment.kubernetesStorageProvisioner === undefined || deployment.kubernetesStorageProvisioner === "")) {
+            deploymentError.push("Storage Provisioner/Storage Class Name cannot be empty");
+        }
+
+        // return error response
+        if (deploymentError.length > 0) {
+            console.log(deploymentError);
+            return res.status(406).send(deploymentError);
```
Comment (rohithteja-cmi) [inline utility/jsonToJdl.js:178]: 400 Bad Request

**Review by rohithteja-cmi — CHANGES_REQUESTED** [review]

File: utility/jsonToJdl.js (hunk 5 above)
Comment (craxkumar) [reply utility/jsonToJdl.js:178]: like empty string/undefined value should not be accepted right ? so is 406(Not acceptable) not appropriate ?

**Review by rohithteja-cmi — APPROVED** [review]

**Pull request merged**